Tests.Unit.Core.ApplicationServices/ApplicationServices/Handlers/HandleUserBeingDeletedTests.cs

```diff
@@ -0,0 +1,260 @@
+        private readonly Mock<IRoleAssignmentService<ItContractRight, ItContractRole, ItContract>> _itContractRightServiceMock;
+        private readonly Mock<IRoleAssignmentService<ItSystemRight, ItSystemRole, ItSystemUsage>> _itSystemRightServiceMock;
+        private readonly Mock<IRoleAssignmentService<ItProjectRight, ItProjectRole, ItProject>> _itProjectRightServiceMock;
+        private readonly Mock<IRoleAssignmentService<OrganizationUnitRight, OrganizationUnitRole, OrganizationUnit>> _organizationUnitRightServiceMock;
+        private readonly Mock<ISsoUserIdentityRepository> _ssoUserIdentityRepository;
+        private readonly HandleUserBeingDeleted _sut;
+
+        public HandleUserBeingDeletedTests()
+        {
+            _dprApplicationServiceMock = new Mock<IDataProcessingRegistrationApplicationService>();
+            _organizationRightsServiceMock = new Mock<IOrganizationRightsService>();
+            _itContractRightServiceMock = new Mock<IRoleAssignmentService<ItContractRight, ItContractRole, ItContract>>();
+            _itSystemRightServiceMock = new Mock<IRoleAssignmentService<ItSystemRight, ItSystemRole, ItSystemUsage>>();
+            _itProjectRightServiceMock = new Mock<IRoleAssignmentService<ItProjectRight, ItProjectRole, ItProject>>();
+            _organizationUnitRightServiceMock = new Mock<IRoleAssignmentService<OrganizationUnitRight, OrganizationUnitRole, OrganizationUnit>>();
+            _ssoUserIdentityRepository = new Mock<ISsoUserIdentityRepository>();
+
+            _sut = new HandleUserBeingDeleted(_dprApplicationServiceMock.Object,
+                _organizationRightsServiceMock.Object,
+                _itContractRightServiceMock.Object,
+                _itSystemRightServiceMock.Object,
+                _itProjectRightServiceMock.Object,
+                _organizationUnitRightServiceMock.Object,
+                _ssoUserIdentityRepository.Object);
+        }
+
+        [Fact]
+        public void Handle_Deletes_All_User_Rights()
+        {
+            var roleId = A<int>();
+            var organizationRole = A<OrganizationRole>();
+            var user = SetupUser(roleId, organizationRole);
+            var isSuccessExpected = true;
+            SetupUserDprMocking(user, roleId, isSuccessExpected);
+            SetupUserOrganizationRightsMocking(user, organizationRole, isSuccessExpected);
+            SetupUserItContractRightsMocking(user, roleId, isSuccessExpected);
+            SetupUserItSystemRightsMocking(user, roleId, isSuccessExpected);
+            SetupUserItProjectRightsMocking(user, roleId, isSuccessExpected);
+            SetupUserOrganizationUnitRightsMocking(user, roleId, isSuccessExpected);
+            SetupSsoIdentitiesMocking(user);
+
+            _sut.Handle(new EntityBeingDeletedEvent<User>(user));
+
+            _dprApplicationServiceMock.Verify(x => x.RemoveRole(It.IsAny<int>(), roleId, user.Id), Times.AtLeastOnce);
```

Comment:
please be more precise in your assertions.
You want to make sure that 
- the excact items from the collections were removed and only them

Samme comment for the remaining assertions.